### ISSUE TITLE
Question about pSMAC3

### ISSUE THREAD

**Issue by ffalfred** [automl/SMAC3#302]
About the pSMAC3. 
I understand I have to launch several commands with the options --shared_model True --input_psmac_dirs smac3-output*, so the smac processes reads the other processes that are being stored in directories with names that start with "smac3-output". But then, at the end of each process, the final parameters are not the same. Should I take as more accurate/better parameters the ones in the last sent process?

Thank you in advance,
Alfred

**Comment by mlindauer** [comment]
Hi Alfred,

> Should I take as more accurate/better parameters the ones in the last sent process?

Please don't use the parameter setting of the last process. Some other processes may have found much better parameter settings. SMAC is still a randomized process.

If you have only a few instances and the final parameter settings of all SMAC processes were evaluated on all (training) instances, you can simply use the best setting.

If you have many instances and SMAC has not evaluated the final parameter setting on all instances, you have to validate all parameter settings on all training instances to determine the best setting.

**Comment by ffalfred** [comment]
Sorry, I am a bit lost with some terminology you use and it is used in the manual: what do you refer as training instances or instances in general? 
I understand instances as each part of the data?

**Comment by mlindauer** [comment]
In contrast to standard hyperparameter optimization, SMAC optimizes an arbitrary performance metric (e.g., runtime or error) across a set of instances. In our terminology, an instance can be a CNF instance of the SAT problem, or a split of a 10-fold CV in ML.
If you simply use SMAC as a hyperparameter optimizer without CV (or you use the f_min interface), you have only a single instance. Thus, you don't need to care about the concept of instances.

Some more details can be found in our best practice paper for algorithm configuration:
https://arxiv.org/pdf/1705.06058.pdf

**Comment by ffalfred** [comment]
I understand now the instance definition, thank you.
I was going to use the f_min interface, but as I commented in a previous post, it fails due to be using a GPU. So I was going to use SMAC as a hyperparameter optimizer on the mean of results of the 5-fold CV.

**Comment by mlindauer** [comment]
Can this issue be closed?

**Comment by ffalfred** [comment]
Yes, although I just have a little question more about the pSMAC. If its basis is to write and read in a shared file space, does it depend the time when each pSMAC run has been launched?

**Comment by mlindauer** [comment]
Yes, all pSMAC runs should start (nearly) at the same time.

**Comment by ffalfred** [comment]
Okey, thank you!
